atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/knowledge/DefaultKnownAbortedTransactions.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * (c) Copyright 2022 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.transaction.knowledge;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.github.benmanes.caffeine.cache.Weigher;
+import com.palantir.atlasdb.AtlasDbConstants;
+import com.palantir.atlasdb.transaction.knowledge.KnownConcludedTransactions.Consistency;
+import java.util.Comparator;
+import java.util.Set;
+import org.checkerframework.checker.index.qual.NonNegative;
+
+public class DefaultKnownAbortedTransactions implements KnownAbortedTransactions {
+    private final KnownConcludedTransactions knownConcludedTransactions;
+    private final AbortedTimestampStore abortedTimestampStore;
+    private final Cache<Long, Set<Long>> reliableCache;
+    private final Cache<Long, Set<Long>> softCache;
+
+    public DefaultKnownAbortedTransactions(
```

Comment:
Some thoughts:

There should always be at most one bucket that cannot be reliably cached. Let's keep these in a separate `ConcurrentHashSet` or something.
When we get a query:
- if it's in one of the cached buckets, we just return
- if it's in the funny bucket, since `isKnownAborted` must be only be called when it is known to be concluded, we can compare the queried timestamp to the greatest known committed since the last update for this bucket. If it is greater, we can update the state of this one set with the diff since last update and then return
- if it's in a greater bucket, then we likely want to update the funny bucket to the end, push it into the cache, reset the funny set, and then decide for this ts what to do